src/Microsoft.DocAsCode.Build.Common/YamlHtmlPart.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.DocAsCode.Build.Common
+{
+    using System.Collections.Generic;
+
+    using HtmlAgilityPack;
+
+    public class YamlHtmlPart
+    {
+        public HtmlDocument Doc { get; set; }
+        public int StartLine { get; set; }
+        public int EndLine { get; set; }
+
+        public static IList<YamlHtmlPart> SplitYamlHtml(string html)
+        {
+            var doc = new HtmlDocument();
+            doc.LoadHtml(html);
+            var parts = new List<YamlHtmlPart>();
+
+            var nodes = doc.DocumentNode.SelectNodes("//yamlheader");
+            if (nodes == null)
+            {
+                return parts;
+            }
+
+            foreach (var node in nodes)
+            {
+                var part = new HtmlDocument();
+
+                var startLineStr = node.GetAttributeValue("start", "-1");
+                var endLineStr = node.GetAttributeValue("end", "-1");
```

Comment:
cool advice to make code clean

---
In reply to: [65125851](https://github.com/dotnet/docfx/pull/343#discussion_r65125851) [](ancestors = 65125851)